common/src/main/scala/com/gu/multimedia/storagetier/vidispine/VidispineCommunicator.scala

```diff
@@ -239,6 +236,46 @@ class VidispineCommunicator(config:VidispineConfig) (implicit ec:ExecutionContex
     ))
   }
 
+  def getFilesOfProject(projectId: Int, pageSize: Int = 100): Future[Seq[VSOnlineOutputMessage]] = {
+      recursivelyGetFilesOfProject(projectId, pageSize = pageSize).map(_.collect({case Some(t) => t}))
+  }
+
+  def recursivelyGetFilesOfProject(projectId: Int, start: Int = 1, pageSize: Int = 100, existingResults: Seq[Option[VSOnlineOutputMessage]] = Seq()): Future[Seq[Option[VSOnlineOutputMessage]]] = {
+    getPageOfFilesOfProject(projectId, start, pageSize).flatMap(results => {
+      if (results.isEmpty || start > 10000) {
+        Future(existingResults ++ results)
+      } else {
+        recursivelyGetFilesOfProject(projectId, start + results.size, pageSize, existingResults ++ results)
+      }
+    })
+  }
+
+  def getPageOfFilesOfProject(projectId:Int, currentItem:Int = 1, pageSize: Int = 100): Future[Seq[Option[VSOnlineOutputMessage]]] = {
+    import io.circe.generic.auto._
+
+    val searchResult = callToVidispine[SearchResultDocument](
+      HttpRequest(
+        uri = s"${config.baseUri}/API/search;first=$currentItem;number=$pageSize?content=shape,metadata&tag=original&field=title,gnm_category,gnm_containing_projects,gnm_nearline_id,itemId",
+        method = HttpMethods.PUT,
+        entity = HttpEntity(ContentTypes.`text/xml(UTF-8)`, s"""
+                                                               |<ItemSearchDocument xmlns="http://xml.vidispine.com/schema/vidispine">
+                                                               |  <field>
+                                                               |    <name>gnm_containing_projects</name>
+                                                               |    <value>$projectId</value>
+                                                               |  </field>
+                                                               |  <intervals>generic</intervals>
+                                                               |</ItemSearchDocument>
+                                                               |""".stripMargin)))
```

Comment:
Probably not worth changing this now, but FYI you can actually type XML _literals_ into scala:

```scala
val doc = <ItemSearchDocument xmlns="">
```
etc.

That gives you an XML node type that you can serialize by using `.toString` - helps because if you get the syntax wrong through a typo it will not compile

Reply:
Oh, very neat!